### ISSUE TITLE
fix channel_force_closure anchor itest flake

### ISSUE THREAD

**Issue by cfromknecht** [lightningnetwork/lnd#4555]
Another flake I observe fairly often (taken from neutrino itests):
```
=== RUN   TestLightningNetworkDaemon/channel_force_closure/committype=anchors
    lnd_test.go:115: Error outside of test: *errors.errorString failed to find commitment in miner mempool: wanted 2, found 1 txs in mempool: [aab923198895a4393bf2912332e44809b3c8f43f56a8528917b99ca042de2d54]
        /home/travis/gopath/src/github.com/lightningnetwork/lnd/lntest/itest/lnd_test.go:3646 (0xdf4469)
        	channelForceClosureTest: t.Fatalf("failed to find commitment in miner mempool: %v", err)
        /home/travis/gopath/src/github.com/lightningnetwork/lnd/lntest/itest/lnd_test.go:3441 (0xe47838)
        	testChannelForceClosure.func1: channelForceClosureTest(
        /home/travis/.gimme/versions/go1.14.7.linux.amd64/src/testing/testing.go:1039 (0x50b9bc)
        	tRunner: fn(t)
        /home/travis/.gimme/versions/go1.14.7.linux.amd64/src/runtime/asm_amd64.s:1373 (0x469401)
        	goexit: BYTE	$0x90	// NOP
```
Here again, could be a missing `WaitPredicate`

**Comment by Roasbeef** [comment]
Does this still occur?

**Comment by Roasbeef** [comment]
A newer instance: 
```
=== RUN   TestLightningNetworkDaemon/06-of-82/neutrino/unconfirmed_channel_funding
    test_harness.go:91: Error outside of test: *errors.errorString unable to find Carol's force close tx in mempool: wanted 2, found 1 txs in mempool: [eb083d362731f2c1afe855930d1bfaa8a69688496cf2c66b347634b3b556cfbe]
        /home/travis/gopath/src/github.com/lightningnetwork/lnd/lntest/itest/lnd_test.go:7940 (0xf57b8c)
        	assertDLPExecuted: t.Fatalf("unable to find Carol's force close tx in mempool: %v",
        /home/travis/gopath/src/github.com/lightningnetwork/lnd/lntest/itest/lnd_channel_backup_test.go:1090 (0xef0b27)
        	testChanRestoreScenario: assertDLPExecuted(
        /home/travis/gopath/src/github.com/lightningnetwork/lnd/lntest/itest/lnd_channel_backup_test.go:408 (0xf78abb)
        	testChannelBackupRestore.func10: testChanRestoreScenario(h, net, &testCase, password)
        /home/travis/.gimme/versions/go1.16.3.linux.amd64/src/testing/testing.go:1193 (0x51c72f)
        	tRunner: fn(t)
        /home/travis/.gimme/versions/go1.16.3.linux.amd64/src/runtime/asm_amd64.s:1371 (0x471c01)
        	goexit: BYTE	$0x90	// NOP
=== CONT  TestLightningNetworkDaemon
    lnd_test.go:11710: Failure time: 2021-07-07 06:09:23.372
--- FAIL: TestLightningNetworkDaemon (949.37s)
    --- PASS: TestLightningNetworkDaemon/41-of-82/neutrino/switch_offline_delivery_persistence (40.45s)
    --- PASS: TestLightningNetworkDaemon/42-of-82/neutrino/switch_offline_delivery_outgoing_offline (29.12s)
    --- PASS: TestLightningNetworkDaemon/43-of-82/neutrino/revoked_uncooperative_close_retribution (17.85s)
    --- PASS: TestLightningNetworkDaemon/44-of-82/neutrino/failing_link (18.52s)
    --- PASS: TestLightningNetworkDaemon/45-of-82/neutrino/garbage_collect_link_nodes (78.41s)
    --- PASS: TestLightningNetworkDaemon/46-of-82/neutrino/abandonchannel (17.06s)
    --- PASS: TestLightningNetworkDaemon/47-of-82/neutrino/revoked_uncooperative_close_retribution_zero_value_remote_output (18.90s)
    --- PASS: TestLightningNetworkDaemon/48-of-82/neutrino/revoked_uncooperative_close_retribution_remote_hodl (20.48s)
    --- PASS: TestLightningNetworkDaemon/49-of-82/neutrino/revoked_uncooperative_close_retribution_altruist_watchtower (56.00s)
        --- PASS: TestLightningNetworkDaemon/49-of-82/neutrino/revoked_uncooperative_close_retribution_altruist_watchtower/anchors (26.33s)
        --- PASS: TestLightningNetworkDaemon/49-of-82/neutrino/revoked_uncooperative_close_retribution_altruist_watchtower/legacy (26.33s)
    --- PASS: TestLightningNetworkDaemon/50-of-82/neutrino/data_loss_protection (66.44s)
    --- PASS: TestLightningNetworkDaemon/51-of-82/neutrino/query_routes (25.91s)
    --- PASS: TestLightningNetworkDaemon/52-of-82/neutrino/route_fee_cutoff (45.28s)
    --- PASS: TestLightningNetworkDaemon/53-of-82/neutrino/send_update_disable_channel (77.72s)
    --- PASS: TestLightningNetworkDaemon/54-of-82/neutrino/streaming_channel_backup_update (45.33s)
    --- PASS: TestLightningNetworkDaemon/55-of-82/neutrino/export_channel_backup (25.77s)
    --- FAIL: TestLightningNetworkDaemon/56-of-82/neutrino/channel_backup_restore (362.25s)
        --- PASS: TestLightningNetworkDaemon/56-of-82/neutrino/channel_backup_restore/restore_from_RPC_backup (40.17s)
        --- PASS: TestLightningNetworkDaemon/56-of-82/neutrino/channel_backup_restore/restore_from_backup_file (38.49s)
        --- PASS: TestLightningNetworkDaemon/56-of-82/neutrino/channel_backup_restore/restore_during_creation (38.43s)
        --- PASS: TestLightningNetworkDaemon/56-of-82/neutrino/channel_backup_restore/restore_during_unlock (40.16s)
        --- PASS: TestLightningNetworkDaemon/56-of-82/neutrino/channel_backup_restore/restore_from_backup_file_twice (38.47s)
        --- PASS: TestLightningNetworkDaemon/56-of-82/neutrino/channel_backup_restore/restore_unconfirmed_channel_file (38.54s)
        --- PASS: TestLightningNetworkDaemon/56-of-82/neutrino/channel_backup_restore/restore_unconfirmed_channel_RPC (38.48s)
        --- FAIL: TestLightningNetworkDaemon/56-of-82/neutrino/channel_backup_restore/restore_from_backup_file_anchors (80.06s)
```

**Comment by yyforyongyu** [comment]
Fixed by #5509 ?

**Comment by Roasbeef** [comment]
Fixed by https://github.com/lightningnetwork/lnd/pull/5509
